src/components/SingleArticle.jsx

```diff
@@ -10,26 +10,41 @@ export default function SingleArticle() {
   const { article_id } = useParams();
   const [article, setArticle] = useState({});
   const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
   const [votes, setVotes] = useState(0);
 
   useEffect(() => {
     getArticleById(article_id).then((response) => {
       setArticle(response.data.article);
       setVotes(response.data.article.votes);
       setIsLoading(false);
+      setError(null);
     });
   }, []);
 
   const handleUpvote = () => {
     setVotes(votes + 1);
-    patchArticleUpvote(article_id);
+    patchArticleUpvote(article_id)
+      .then((response) => {
+        setError(null);
+      })
+      .catch((err) => {
+        setError("Voting failed, please try again later");
+      });
   };
 
   const handleDownvote = () => {
     setVotes(votes - 1);
-    patchArticleDownvote(article_id);
+    patchArticleDownvote(article_id)
+      .then((response) => {
+        setError(null);
+      })
+      .catch((err) => {
+        setError("Voting failed, please try again later");
+      });
```

Comment:
You have two functions here which do very similar, if you had one function that handles both buttons you could pass it an argument of 1 or -1 and then use that in a single function. You could do the same with the axios requests too!